src/blk/spdk/NVMEDevice.cc

```diff
@@ -90,12 +94,34 @@ class SharedDriverData {
         ns(ns_) {
     block_size = spdk_nvme_ns_get_extended_sector_size(ns);
     size = spdk_nvme_ns_get_size(ns);
+    if (trid.trtype == SPDK_NVME_TRANSPORT_PCIE) {
+      return;
+    }
+
+    if (!admin_thread.joinable()) {
```

Comment:
as `SharedDriverData` is just constructed, `admin_thread.joinable()` should always evaluate to `true`, am i right?

Reply:
ack